stardis/radiation_field/radiation_field_solvers/base.py

```diff
@@ -312,15 +360,42 @@ def raytrace(stellar_model, stellar_radiation_field, no_of_thetas=20, n_threads=
         )
 
     else:  # Parallel threaded
-        for theta in thetas:
-            weight = 2 * np.pi * dtheta * np.sin(theta) * np.cos(theta)
-            stellar_radiation_field.F_nu += weight * single_theta_trace_parallel(
-                stellar_model.geometry.dist_to_next_depth_point,
+        for theta_index, theta in enumerate(thetas):
+            stellar_radiation_field.F_nu += weights[
+                theta_index
+            ] * single_theta_trace_parallel(
+                ray_distances[:, theta_index],
                 stellar_model.temperatures.value.reshape(-1, 1),
                 stellar_radiation_field.opacities.total_alphas,
                 stellar_radiation_field.frequencies,
                 theta,
                 stellar_radiation_field.source_function,
             )
+    if spherical:
+        stellar_radiation_field.F_nu *= photometric_correction
 
     return stellar_radiation_field.F_nu
+
+
+def calculate_spherical_ray(thetas, depth_points_radii):
+    ###NOTE: This will need to be revisited to handle some rays more carefully if they don't go through the star
+    ray_distance_through_layer_by_impact_parameter = np.zeros(
+        (len(depth_points_radii) - 1, len(thetas))
+    )
+
+    for theta_index, theta in enumerate(thetas):
+        b = depth_points_radii[-1] * np.sin(theta)  # impact parameter of the ray
+        ray_z_coordinate_grid = np.sqrt(depth_points_radii**2 - b**2)
+
+        dr = np.diff(depth_points_radii)
+        ds = dr * depth_points_radii[1:] / ray_z_coordinate_grid[1:]
+        ray_distance_through_layer_by_impact_parameter[~np.isnan(ds), theta_index] = ds[
+            ~np.isnan(ds)
+        ]
+
+        # ray_distance = np.diff(ray_z_coordinate_grid)
```

Comment:
Do you plan to keep this code?

Reply:
Nope! I had already removed it in the readability cleanup commit I just forgot to push to the branch.